rlp/rlp.go

```diff
@@ -84,8 +84,10 @@ func Decode(input []byte) *Item {
 		n := input[0] - str1H
 		return &Item{D: input[1:n]}
 	case input[0] <= strNH:
-		headerSize := 1 + input[0] - str55H
-		return &Item{D: input[headerSize:]}
+		i := int(1 + (input[0] - str55H)) //head length
+		n := 1                            //header byte
+		n += decodeLength(str55H, input)
+		return &Item{D: input[i:n]}
```

Comment:
correct me if i'm wrong here, but if `n` is the length of the string starting from `i`, then should it be `input[i:i+n]`?

Reply:
This is a little confusing. `decodeLength` returns the length including the header. I thought about making `decodeLength` return 2 ints -- the header and payload.

Reply:
Gotcha, thanks for clarifying